src/CoreBundle/Maintenance/MaintenanceMode/MaintenanceModeService.php

```diff
@@ -34,6 +34,17 @@ public function __construct(Connection $connection, CacheInterface $cache)
         $this->cache = $cache;
     }
 
+    public function isActivated(): bool
+    {
+        if (true === \TdbCmsConfig::GetInstance()->fieldShutdownWebsites) {
+            clearstatcache(true, PATH_MAINTENANCE_MODE_MARKER);
```

Comment:
We should add a comment that describes why we clear the stat cache - I think this is not self-explanatory. And as the maintenance mode functionality is quite scattered across the code base (unavoidably), we should also write some RST documentation describing our thoughts (including multi-node support).

Reply:
ok ok